### PR TITLE
adding a simple URL conversion tool

### PR DESCRIPTION
**Pull request by davidnuescheler**
this creates a little form on the `project-helix.page` website where people can paste a URL pointing to a publicly visible `.md` in github and redirects them to the corresponding helix pages URL

### CODE REVIEW

**Comment by davidnuescheler** [comment]
in a weird self referential way, test it here, minus the css.
https://url-tool--helix-pages--adobe.project-helix.page/index.html

File: index.md [hunk 1]
```diff
@@ -2,7 +2,37 @@
...
 Example: [https://helix-home-adobe.project-helix.page/README.html](https://helix-home-adobe.project-helix.page/README.html)
+
+---
+
+### Try it...
+Simply paste a GitHub URL to a publicly visible Markdown file (`.md`) here... 
+
+<script>
+function takeMeThere() {
+    let giturl=document.getElementById('giturl').value;
```
Comment (kptdobe) [inline index.md]:
```suggestion
    const giturl = document.getElementById('giturl').value;
```

File: index.md [hunk 2]
```diff
@@ -2,7 +2,37 @@
...
+    let resegs=/(?<!\?.+)(?<=\/)[\w-]+(?=[/\r\n?]|$)/g;
```
Comment (kptdobe) [inline index.md]:
```suggestion
    const resegs = /(?<!\?.+)(?<=\/)[\w-]+(?=[/\r\n?]|$)/g;
```

File: index.md [hunk 3]
```diff
@@ -2,7 +2,37 @@
...
+    let segments = [...giturl.matchAll(resegs)];
```
Comment (kptdobe) [inline index.md]:
```suggestion
    const segments = [...giturl.matchAll(resegs)];
```

File: index.md [hunk 4]
```diff
@@ -2,7 +2,37 @@
...
+    let separator = '-';
+    const user = segments[0][0];
+    const repo = segments[1][0];
+    const branch = segments[3][0];
+    console.log (segments);
+    const path = giturl.substr(segments[3].index+branch.length, giturl.length - (segments[3].index+branch.length) - 3);
```
Comment (kptdobe) [inline index.md]:
```suggestion
    const path = giturl.substr(segments[3].index + branch.length, giturl.length - (segments[3].index + branch.length) - 3);
```

File: index.md [hunk 5]
```diff
@@ -2,7 +2,37 @@
...
+<script>
+function takeMeThere() {
+    let giturl=document.getElementById('giturl').value;
+    let resegs=/(?<!\?.+)(?<=\/)[\w-]+(?=[/\r\n?]|$)/g;
+    let segments = [...giturl.matchAll(resegs)];
+    let separator = '-';
+    const user = segments[0][0];
+    const repo = segments[1][0];
+    const branch = segments[3][0];
+    console.log (segments);
```
Comment (kptdobe) [inline index.md]:
```suggestion
```

File: index.md [hunk 6]
```diff
@@ -2,7 +2,37 @@
...
+    const path = giturl.substr(segments[3].index+branch.length, giturl.length - (segments[3].index+branch.length) - 3);
+    if (user.indexOf('-')>=0 || branch!='master') {
```
Comment (kptdobe) [inline index.md]:
```suggestion
    if (user.indexOf('-') >= 0 || branch != 'master') {
```

File: index.md [hunk 7]
```diff
@@ -2,7 +2,37 @@
...
+        separator = '--';
+    }
+    const branchprefix=(branch=='master'?"":branch+separator);
```
Comment (kptdobe) [inline index.md]:
```suggestion
    const branchprefix = (branch=='master' ? '' : branch+separator);
```

File: index.md [hunk 8]
```diff
@@ -2,7 +2,37 @@
...
+    const url=`https://${branchprefix}${repo}${separator}${user}.project-helix.page${path}.html`;
```
Comment (kptdobe) [inline index.md]:
```suggestion
    const url = `https://${branchprefix}${repo}${separator}${user}.project-helix.page${path}.html`;
```

File: index.md [hunk 9]
```diff
@@ -2,7 +2,37 @@
...
+    window.location=url;
```
Comment (kptdobe) [inline index.md]:
```suggestion
    window.location = url;
```

File: index.md [hunk 10]
```diff
@@ -2,7 +2,37 @@
 
 Welcome to Helix Pages!
 
-To use it, change the current URL to `https://<repo>-<owner>.project-helix.page`.
+To use it, change the current URL    to `https://<repo>-<owner>.project-helix.page`.
```
Comment (kptdobe) [inline index.md]:
```suggestion
To use it, change the current URL to `https://<repo>-<owner>.project-helix.page`.
```

**Pull request merged**